api/src/main/java/org/apache/iceberg/transforms/Transform.java

```diff
@@ -58,6 +58,29 @@
    */
   Type getResultType(Type sourceType);
 
+  /**
+   * Whether the transform preserves the order of values (is monotonic).
+   * <p>
+   * A transform preserves order for values when for any given a and b, if a &lt; b then apply(a) &lt;= apply(b).
```

Comment:
Wouldn't it also be the case that a = b; then apply(a) = apply(b)?

Reply:
If a = b, then apply(a) should always equal apply(b). This is concerned with the case where a and b are not equal.

For example, a year transform has this property. year(2020-12-31) = 2020 and year(2021-01-01) = 2021. Because the first date comes before the second, its year must be before (or the same as) any date after it.